src/util_libc.rs

```diff
@@ -116,3 +116,21 @@ impl LazyFd {
         }
     }
 }
+
+cfg_if! {
+    if #[cfg(any(target_os = "linux", target_os = "emscripten"))] {
+        use libc::open64 as open;
+    } else {
+        use libc::open;
+    }
+}
+
+// SAFETY: path must be null terminated, FD must be manually closed.
+pub unsafe fn open_readonly(path: &str) -> Option<libc::c_int> {
+    // We don't care about Linux OSes too old to support O_CLOEXEC.
+    let fd = open(path.as_ptr() as *mut _, libc::O_RDONLY | libc::O_CLOEXEC);
```

Comment:
It may be worth to add a `debug_assert!` which will check that `path` is null terminated. Of course it's not necessary, but if it's possible to check safety conditions, personally I usually add such asserts to be extra safe.

>We don't care about Linux OSes too old to support O_CLOEXEC.

Strictly speaking Rust supports Linux kernel from [2.16.18 and later](https://forge.rust-lang.org/platform-support.html), while `O_CLOEXEC` is available only from 2.6.23. To be safe I think we should run it by the lang team. I will try to ask about this in the rust-lang PR.